codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/parse/CborParserGenerator.kt

```diff
@@ -216,7 +247,7 @@ class CborParserGenerator(
                         val callBuilderSetMemberFieldWritable =
                             writable {
                                 withBlock("builder.${member.setterName()}(", ")") {
-                                    conditionalBlock("Some(", ")", symbolProvider.toSymbol(member).isOptional()) {
+                                    conditionalBlock("Some(", ")", codegenTarget != CodegenTarget.SERVER || symbolProvider.toSymbol(member).isOptional()) {
```

Comment:
Can't you use the other setter client builders generate? The one prefixed with `set_` (e.g. `set_dense_struct_map`) always takes `Option`s, while the one unprefixed (e.g. `dense_struct_map`) takes the actual type as modeled, so you wouldn't need `codegenTarget` check here.

Reply:
The short answer is no AFAIK. For the client, unprefixed setters for collections don't behave as you described, unfortunately. For instance, here is what `default_map` and `set_default_map` look like:
```
    /// Adds a key-value pair to `default_map`.
    ///
    /// To override the contents of this collection use [`set_default_map`](Self::set_default_map).
    ///
    pub fn default_map(mut self, k: impl ::std::convert::Into<::std::string::String>, v: impl ::std::convert::Into<::std::string::String>) -> Self {
        let mut hash_map = self.default_map.unwrap_or_default();
        hash_map.insert(k.into(), v.into());
        self.default_map = ::std::option::Option::Some(hash_map);
        self
    }
    #[allow(missing_docs)] // documentation missing in model
    pub fn set_default_map(
        mut self,
        input: ::std::option::Option<::std::collections::HashMap<::std::string::String, ::std::string::String>>,
    ) -> Self {
        self.default_map = input;
        self
    }
```
As you can see, the unprefixed setter only inserts a new key-value and is not a sibling method of the prefixed counterpart (this has been done [here](https://github.com/smithy-lang/smithy-rs/blob/1223f61e10c82b5b8172ad9e18cd6cd558203751/codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/BuilderGenerator.kt#L405) on the client side for a long time).

Reply:
4d24ece will make this behavior configurable through a `CborParserGenerator`'s constructor argument.